src/bundle/Resources/public/js/scripts/fieldType/base/base-field.js

```diff
@@ -0,0 +1,90 @@
+(function (global, doc) {
+    const eZ = global.eZ = global.eZ || {};
+
+    eZ.baseFieldValidator = class BaseFieldValidator {
+        constructor(config) {
+            this.classInvalid = config.classInvalid;
+            this.eventsMap = config.eventsMap;
+            this.fieldSelector = config.fieldSelector;
+        }
+
+        attachEvent(eventName, selector, callback) {
+            [...doc.querySelectorAll(selector)].forEach(item => item.addEventListener(eventName, callback, false));
+        }
+
+        removeEvent(eventName, selector, callback) {
+            [...doc.querySelectorAll(selector)].forEach(item => item.removeEventListener(eventName, callback, false));
+        }
+
+        findStateNodes(fieldNode, input, selectors) {
```

Comment:
I'm not sure it does the thing the method name points to. It creates a new flattened array of elements found by each selector.

Reply:
I think better name could be: `findNodesBySelectors`